File: TP1cpp/Produk.cpp
#include <iostream>
#include <string>
#include <iomanip>      //untuk setw di tabel
using namespace std;

// Class Produk untuk menyimpan data barang elektronik
class Produk {
private:
    string id;              // Atribut unik produk
    string nama;            // Atribut nama untuk produk
    string merek;           // Atribut untuk merek produk
    string kategori;        // Atribut kategori untuk produk
    int harga;              // Atribut harga produk
    int stok;               // stok produk

public:
    // Constructor
    Produk(string id = "", string nama = "", string merek = "", string kategori = "", int harga = 0,  int stok = 0) {
        this->id = id;                  // Isi Id
        this->nama = nama;              // Isi nama
        this->merek = merek;            // Isi merek
        this->kategori = kategori;      // Isi kategori
        this->harga = harga;            // Isi harga
        this->stok = stok;              // Isi stok
    }

    // Destructor
    ~Produk() {

    }

    // Setter untuk update data
    void setNama(string nama) { this->nama = nama; }
    void setMerek(string merek) { this->merek = merek; }
    void setKategori(string kategori) { this->kategori = kategori; }
    void setHarga(int harga) { this->harga = harga; }
    void setStok(int stok) { this->stok = stok; }


    // Getter untuk membaca data privat
    string getId() { return id; }
    string getNama() { return nama; }
    string getMerek() { return merek; }
    string getKategori() { return kategori; }
    int getHarga() { return harga; }
    int getStok() { return stok; }

    void tampil(int no) {
        cout << setw(3) << right << no << " | "
             << setw(8) << left << id << " | "
             << setw(20) << left << nama << " | "
             << setw(15) << left << merek << " | "
             << setw(15) << left << kategori << " | "
             << setw(8) << right << harga << " | "
             << setw(11) << right << stok << " | " 
             << endl;
    }

    
};

File: Toko_Elektronik/Produk.php
<?php
// Produk.php
class Produk {
    private $id;
    private $nama;
    private $merek;
    private $kategori;
    private $harga;
    private $stok;
    private $gambar; // path lokal (misal: uploads/nama_file.jpg)

    public function __construct($id = "", $nama = "", $merek = "", $kategori = "", $harga = 0, $stok = 0, $gambar = "") {
        $this->id = (string)$id;
        $this->nama = (string)$nama;
        $this->merek = (string)$merek;
        $this->kategori = (string)$kategori;
        $this->harga = (int)$harga;
        $this->stok = (int)$stok;
        $this->gambar = (string)$gambar;
    }

    // getters
    public function getId() { return $this->id; }
    public function getNama() { return $this->nama; }
    public function getMerek() { return $this->merek; }
    public function getKategori() { return $this->kategori; }
    public function getHarga() { return $this->harga; }
    public function getStok() { return $this->stok; }
    public function getGambar() { return $this->gambar; }

    // setters
    public function setNama($v) { $this->nama = (string)$v; }
    public function setMerek($v) { $this->merek = (string)$v; }
    public function setKategori($v) { $this->kategori = (string)$v; }
    public function setHarga($v) { $this->harga = (int)$v; }
    public function setStok($v) { $this->stok = (int)$v; }
    public function setGambar($v) { $this->gambar = (string)$v; }
}
